purchase_product_barcode/models/purchase_order_line.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    barcode = fields.Char(
+        related='product_id.barcode',
+        string='Barcode',
+        copy=False,
+    )
```

Comment:
Considering about readonly attribute to this field.

Reply:
Yes, please make the field readonly.